examples/server/server.cpp

```diff
@@ -1321,17 +1321,24 @@ struct server_slot {
             && are_lora_equal(lora, other_slot.lora);
     }
 
+    //  There are two caps on the budge of a single request:
+    //  * [params.n_predict]
+    //  * [global_params.n_predict]
+    // This function returns true if the request is not limited by either of them.
     bool has_budget(const common_params & global_params) {
         if (params.n_predict == -1 && global_params.n_predict == -1) {
             return true; // limitless
         }
+        n_remaining = INT32_MAX;
 
-        n_remaining = -1;
+        // The request or server have finite limits on the number of tokens to generate.
+        if ((params.n_predict != -1 && params.n_predict != -2) || (global_params.n_predict  != -1 && global_params.n_predict != -2)) {
```

Comment:
For readability, can we extract this `params.n_predict != -1 && params.n_predict != -2` and also the other group into a bool?

Example

```
bool req_has_finite_limit = params.n_predict != -1 && params.n_predict != -2;
bool srv_has_finite_limit = global_params.n_predict != -1 && global_params.n_predict != -2;
```

Reply:
`-2` is also finite.

For readibility we can change to `params.n_predict >= 0`